Attempt to invoke virtual method 'an droid.content.res.Resources android.view.View.getResources()' on a null object reference

**Summary:** 

Got a crash when starting the app.

**Steps to reproduce:** 

Not easy to reproduce, but second time today.

**System logs:**

```
11-29 22:39:35.025 23369 23517 D NoSync  : Called fsync                                                                                
11-29 22:39:35.029 23592 23809 I native  : flash-index.cc:1605 Flash index sort hb 11198 prefix-hits 49416 -> 33239 at 539.188ms
11-29 22:39:35.101 23592 11637 I iu.FingerprintManager: Start processing media store URI: content://media/external/video/media            
11-29 22:39:35.169 12009 12009 D NearbyFragment: On nearby tab selected                                        
11-29 22:39:35.207   794   456 I GnssLocationProvider: WakeLock acquired by sendMessage(3, 0, com.android.server.location.GnssLocationProvider$GpsRequest@1a6fa98)
11-29 22:39:35.210 12009 12009 D NearbyFragment: checking GPS                                                 
11-29 22:39:35.210 12009 12009 D NearbyFragment: GPS is enabled                                                                                                                                                                    
11-29 22:39:35.211 12009 12009 D NearbyFragment: Checking location permission                                                                                                                                                                           
11-29 22:39:35.212 12009 12009 D NearbyFragment: Refreshing nearby places                                                                                           
11-29 22:39:35.214 12009 12009 D NearbyFragment: Skipping update of nearby places as location is unavailable 
11-29 22:39:35.217 12009 12009 D AndroidRuntime: Shutting down VM                                             
11-29 22:39:35.218 12009 12009 E AndroidRuntime: FATAL EXCEPTION: main                      
11-29 22:39:35.218 12009 12009 E AndroidRuntime: Process: fr.free.nrw.commons, PID: 12009   
11-29 22:39:35.218 12009 12009 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'an
droid.content.res.Resources android.view.View.getResources()' on a null object reference                                                                                                                          
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2684)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)                        
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread.-wrap12(ActivityThread.java)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)                                                     
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:102)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:154)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6186)                             
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
11-29 22:39:35.218 12009 12009 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.view.View.getResources()' on a null object reference
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.support.design.widget.Snackbar.make(Snackbar.java:170)                                                                           
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at fr.free.nrw.commons.nearby.NearbyFragment.addNetworkBroadcastReceiver(NearbyFragment.java:596)                                                                                                    
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at fr.free.nrw.commons.nearby.NearbyFragment.performNearbyOperations(NearbyFragment.java:639)                                               
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at fr.free.nrw.commons.nearby.NearbyFragment.onTabSelected(NearbyFragment.java:627)                
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at fr.free.nrw.commons.contributions.MainActivity.onCreate(MainActivity.java:87)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:6684)                                                                                 
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)                                             
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
11-29 22:39:35.218 12009 12009 E AndroidRuntime:        ... 9 more                                                                                                                                                           
11-29 22:39:35.238   794  3630 W ActivityManager:   Force finishing activity fr.free.nrw.commons/.contributions.MainActivity
```

**Device and Android version:** 

LineageOS Android 7.1.2
 
**Commons app version:** 

2.9.0 from Play Store

--- a/app/src/main/java/fr/free/nrw/commons/auth/AuthenticatedActivity.java
+++ b/app/src/main/java/fr/free/nrw/commons/auth/AuthenticatedActivity.java
@@ -64,7 +64,7 @@ public abstract class AuthenticatedActivity extends NavigationBaseActivity {
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .filter(result -> result)
-                .subscribe(result -> ViewUtil.showSnackbar(findViewById(android.R.id.content), R.string.block_notification)
+                .subscribe(result -> ViewUtil.showShortSnackbar(findViewById(android.R.id.content), R.string.block_notification)
                 );
     }
 }

--- a/app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/bookmarks/pictures/BookmarkPicturesFragment.java
@@ -129,7 +129,7 @@ public class BookmarkPicturesFragment extends DaggerFragment {
             statusTextView.setVisibility(VISIBLE);
             statusTextView.setText(getString(R.string.no_internet));
         } else {
-            ViewUtil.showSnackbar(parentLayout, R.string.no_internet);
+            ViewUtil.showShortSnackbar(parentLayout, R.string.no_internet);
         }
     }
 
@@ -140,7 +140,7 @@ public class BookmarkPicturesFragment extends DaggerFragment {
     private void handleError(Throwable throwable) {
         Timber.e(throwable, "Error occurred while loading images inside a category");
         try{
-            ViewUtil.showSnackbar(parentLayout, R.string.error_loading_images);
+            ViewUtil.showShortSnackbar(parentLayout, R.string.error_loading_images);
             initErrorView();
         }catch (Exception e){
             e.printStackTrace();

--- a/app/src/main/java/fr/free/nrw/commons/category/CategoryImagesListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/CategoryImagesListFragment.java
@@ -124,7 +124,7 @@ public class CategoryImagesListFragment extends DaggerFragment {
             statusTextView.setVisibility(VISIBLE);
             statusTextView.setText(getString(R.string.no_internet));
         } else {
-            ViewUtil.showSnackbar(parentLayout, R.string.no_internet);
+            ViewUtil.showShortSnackbar(parentLayout, R.string.no_internet);
         }
     }
 
@@ -135,7 +135,7 @@ public class CategoryImagesListFragment extends DaggerFragment {
     private void handleError(Throwable throwable) {
         Timber.e(throwable, "Error occurred while loading images inside a category");
         try{
-            ViewUtil.showSnackbar(parentLayout, R.string.error_loading_images);
+            ViewUtil.showShortSnackbar(parentLayout, R.string.error_loading_images);
             initErrorView();
         }catch (Exception e){
             e.printStackTrace();

--- a/app/src/main/java/fr/free/nrw/commons/category/SubCategoryListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/category/SubCategoryListFragment.java
@@ -135,10 +135,10 @@ public class SubCategoryListFragment extends CommonsDaggerSupportFragment {
     private void handleError(Throwable throwable) {
         if (!isParentCategory){
             Timber.e(throwable, "Error occurred while loading queried subcategories");
-            ViewUtil.showSnackbar(categoriesRecyclerView,R.string.error_loading_categories);
+            ViewUtil.showShortSnackbar(categoriesRecyclerView,R.string.error_loading_categories);
         }else {
             Timber.e(throwable, "Error occurred while loading queried parentcategories");
-            ViewUtil.showSnackbar(categoriesRecyclerView,R.string.error_loading_categories);
+            ViewUtil.showShortSnackbar(categoriesRecyclerView,R.string.error_loading_categories);
         }
     }
 
@@ -161,6 +161,6 @@ public class SubCategoryListFragment extends CommonsDaggerSupportFragment {
      */
     private void handleNoInternet() {
         progressBar.setVisibility(GONE);
-        ViewUtil.showSnackbar(categoriesRecyclerView, R.string.no_internet);
+        ViewUtil.showShortSnackbar(categoriesRecyclerView, R.string.no_internet);
     }
 }

--- a/app/src/main/java/fr/free/nrw/commons/explore/categories/SearchCategoryFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/explore/categories/SearchCategoryFragment.java
@@ -196,7 +196,7 @@ public class SearchCategoryFragment extends CommonsDaggerSupportFragment {
         Timber.e(throwable, "Error occurred while loading queried categories");
         try {
             initErrorView();
-            ViewUtil.showSnackbar(categoriesRecyclerView, R.string.error_loading_categories);
+            ViewUtil.showShortSnackbar(categoriesRecyclerView, R.string.error_loading_categories);
         }catch (Exception e){
             e.printStackTrace();
         }
@@ -217,6 +217,6 @@ public class SearchCategoryFragment extends CommonsDaggerSupportFragment {
      */
     private void handleNoInternet() {
         progressBar.setVisibility(GONE);
-        ViewUtil.showSnackbar(categoriesRecyclerView, R.string.no_internet);
+        ViewUtil.showShortSnackbar(categoriesRecyclerView, R.string.no_internet);
     }
 }

--- a/app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java
@@ -206,7 +206,7 @@ public class SearchImageFragment extends CommonsDaggerSupportFragment {
     private void handleError(Throwable throwable) {
         Timber.e(throwable, "Error occurred while loading queried images");
         try {
-            ViewUtil.showSnackbar(imagesRecyclerView, R.string.error_loading_images);
+            ViewUtil.showShortSnackbar(imagesRecyclerView, R.string.error_loading_images);
         }catch (Exception e){
             e.printStackTrace();
         }
@@ -226,7 +226,7 @@ public class SearchImageFragment extends CommonsDaggerSupportFragment {
      */
     private void handleNoInternet() {
         progressBar.setVisibility(GONE);
-        ViewUtil.showSnackbar(imagesRecyclerView, R.string.no_internet);
+        ViewUtil.showShortSnackbar(imagesRecyclerView, R.string.no_internet);
     }
 
     /**

--- a/app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/nearby/NearbyFragment.java
@@ -15,7 +15,6 @@ import android.support.design.widget.Snackbar;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v7.app.AlertDialog;
 import android.view.LayoutInflater;
-
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.LinearLayout;
@@ -37,6 +36,7 @@ import fr.free.nrw.commons.di.CommonsDaggerSupportFragment;
 import fr.free.nrw.commons.location.LatLng;
 import fr.free.nrw.commons.location.LocationServiceManager;
 import fr.free.nrw.commons.location.LocationUpdateListener;
+import fr.free.nrw.commons.utils.FragmentUtils;
 import fr.free.nrw.commons.utils.NetworkUtils;
 import fr.free.nrw.commons.utils.UriSerializer;
 import fr.free.nrw.commons.utils.ViewUtil;
@@ -324,7 +324,7 @@ public class NearbyFragment extends CommonsDaggerSupportFragment
         String gsonBoundaryCoordinates = gson.toJson(boundaryCoordinates);
 
         if (placeList.size() == 0) {
-            ViewUtil.showSnackbar(view.findViewById(R.id.container), R.string.no_nearby);
+            ViewUtil.showShortSnackbar(view.findViewById(R.id.container), R.string.no_nearby);
         }
 
         bundle.putString("PlaceList", gsonPlaceList);
@@ -592,14 +592,17 @@ public class NearbyFragment extends CommonsDaggerSupportFragment
     }
 
     private void addNetworkBroadcastReceiver() {
+        if (!FragmentUtils.isFragmentUIActive(this)) {
+            return;
+        }
+        
         IntentFilter intentFilter = new IntentFilter(NETWORK_INTENT_ACTION);
-            snackbar = Snackbar.make(transparentView , R.string.no_internet, Snackbar.LENGTH_INDEFINITE);
+        snackbar = Snackbar.make(transparentView, R.string.no_internet, Snackbar.LENGTH_INDEFINITE);
 
-            broadcastReceiver = new BroadcastReceiver() {
-
+        broadcastReceiver = new BroadcastReceiver() {
             @Override
             public void onReceive(Context context, Intent intent) {
-                if (snackbar != null) {
+                if (snackbar != null && getActivity() != null) {
                     if (NetworkUtils.isInternetConnectionEstablished(getActivity())) {
                         refreshView(LOCATION_SIGNIFICANTLY_CHANGED);
                         snackbar.dismiss();
@@ -609,6 +612,10 @@ public class NearbyFragment extends CommonsDaggerSupportFragment
                 }
             }
         };
+
+        if (getActivity() == null) {
+            return;
+        }
 
         getActivity().registerReceiver(broadcastReceiver, intentFilter);
 

--- a/app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java
+++ b/app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java
@@ -106,7 +106,7 @@ public class NotificationActivity extends NavigationBaseActivity {
                         progressBar.setVisibility(View.GONE);
                     }, throwable -> {
                         Timber.e(throwable, "Error occurred while loading notifications");
-                        ViewUtil.showSnackbar(relativeLayout, R.string.error_notifications);
+                        ViewUtil.showShortSnackbar(relativeLayout, R.string.error_notifications);
                         progressBar.setVisibility(View.GONE);
                     });
         } else {
@@ -123,7 +123,7 @@ public class NotificationActivity extends NavigationBaseActivity {
 
     private void setAdapter(List<Notification> notificationList) {
         if (notificationList == null || notificationList.isEmpty()) {
-            ViewUtil.showSnackbar(relativeLayout, R.string.no_notifications);
+            ViewUtil.showShortSnackbar(relativeLayout, R.string.no_notifications);
             return;
         }
         notificationAdapterFactory = new NotificationAdapterFactory(notification -> {

--- a/app/src/main/java/fr/free/nrw/commons/utils/FragmentUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/FragmentUtils.java
@@ -28,4 +28,8 @@ public class FragmentUtils {
         }
         return false;
     }
+
+    public static boolean isFragmentUIActive(Fragment fragment) {
+        return fragment.getActivity() != null && fragment.isAdded() && !fragment.isDetached() && !fragment.isRemoving();
+    }
 }

--- a/app/src/main/java/fr/free/nrw/commons/utils/FragmentUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/FragmentUtils.java
@@ -29,6 +29,11 @@ public class FragmentUtils {
         return false;
     }
 
+    /**
+     * Utility function to check whether the fragment UI is still active or not
+     * @param fragment
+     * @return
+     */
     public static boolean isFragmentUIActive(Fragment fragment) {
         return fragment.getActivity() != null && fragment.isAdded() && !fragment.isDetached() && !fragment.isRemoving();
     }

--- a/app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/NetworkUtils.java
@@ -11,6 +11,9 @@ public class NetworkUtils {
         ConnectivityManager cm =
                 (ConnectivityManager)context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
 
+        if (cm == null) {
+            return false;
+        }
         NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
         return activeNetwork != null &&
                 activeNetwork.isConnectedOrConnecting();

--- a/app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java
@@ -17,12 +17,20 @@ public class ViewUtil {
     public static final String SHOWCASE_VIEW_ID_2 = "SHOWCASE_VIEW_ID_2";
     public static final String SHOWCASE_VIEW_ID_3 = "SHOWCASE_VIEW_ID_3";
 
-    public static void showSnackbar(View view, int messageResourceId) {
+    public static void showShortSnackbar(View view, int messageResourceId) {
         if (view.getContext() == null) {
             return;
         }
 
         ExecutorUtils.uiExecutor().execute(() -> Snackbar.make(view, messageResourceId, Snackbar.LENGTH_SHORT).show());
+    }
+
+    public static void showIndefiniteSnackbar(View view, int messageResourceId) {
+        if (view.getContext() == null) {
+            return;
+        }
+
+        ExecutorUtils.uiExecutor().execute(() -> Snackbar.make(view, messageResourceId, Snackbar.LENGTH_INDEFINITE).show());
     }
 
     public static void showLongToast(Context context, String text) {

--- a/app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/ViewUtil.java
@@ -17,20 +17,17 @@ public class ViewUtil {
     public static final String SHOWCASE_VIEW_ID_2 = "SHOWCASE_VIEW_ID_2";
     public static final String SHOWCASE_VIEW_ID_3 = "SHOWCASE_VIEW_ID_3";
 
+    /**
+     * Utility function to show short snack bar
+     * @param view
+     * @param messageResourceId
+     */
     public static void showShortSnackbar(View view, int messageResourceId) {
         if (view.getContext() == null) {
             return;
         }
 
         ExecutorUtils.uiExecutor().execute(() -> Snackbar.make(view, messageResourceId, Snackbar.LENGTH_SHORT).show());
-    }
-
-    public static void showIndefiniteSnackbar(View view, int messageResourceId) {
-        if (view.getContext() == null) {
-            return;
-        }
-
-        ExecutorUtils.uiExecutor().execute(() -> Snackbar.make(view, messageResourceId, Snackbar.LENGTH_INDEFINITE).show());
     }
 
     public static void showLongToast(Context context, String text) {

